renderdoc/driver/d3d12/d3d12_counters.cpp

```diff
@@ -228,106 +241,257 @@ vector<CounterResult> D3D12Replay::FetchCounters(const vector<uint32_t> &counter
   heapProps.CreationNodeMask = 1;
   heapProps.VisibleNodeMask = 1;
 
-  D3D12_RESOURCE_DESC bufDesc;
-  bufDesc.Dimension = D3D12_RESOURCE_DIMENSION_BUFFER;
-  bufDesc.Alignment = 0;
-  bufDesc.Width = sizeof(uint64_t) * maxEID * 2;
-  bufDesc.Height = 1;
-  bufDesc.DepthOrArraySize = 1;
-  bufDesc.MipLevels = 1;
-  bufDesc.Format = DXGI_FORMAT_UNKNOWN;
-  bufDesc.SampleDesc.Count = 1;
-  bufDesc.SampleDesc.Quality = 0;
-  bufDesc.Layout = D3D12_TEXTURE_LAYOUT_ROW_MAJOR;
-  bufDesc.Flags = D3D12_RESOURCE_FLAG_NONE;
-
-  ID3D12Resource *readbackBuf;
-  HRESULT hr = m_pDevice->CreateCommittedResource(&heapProps, D3D12_HEAP_FLAG_NONE, &bufDesc,
-                                                  D3D12_RESOURCE_STATE_COPY_DEST, NULL,
-                                                  __uuidof(ID3D12Resource), (void **)&readbackBuf);
+  D3D12_RESOURCE_DESC timerBufDesc;
+  timerBufDesc.Dimension = D3D12_RESOURCE_DIMENSION_BUFFER;
+  timerBufDesc.Alignment = 0;
+  timerBufDesc.Width = sizeof(uint64_t) * maxEID * 2;
+  timerBufDesc.Height = 1;
+  timerBufDesc.DepthOrArraySize = 1;
+  timerBufDesc.MipLevels = 1;
+  timerBufDesc.Format = DXGI_FORMAT_UNKNOWN;
+  timerBufDesc.SampleDesc.Count = 1;
+  timerBufDesc.SampleDesc.Quality = 0;
+  timerBufDesc.Layout = D3D12_TEXTURE_LAYOUT_ROW_MAJOR;
+  timerBufDesc.Flags = D3D12_RESOURCE_FLAG_NONE;
+
+  ID3D12Resource *timerReadbackBuf;
```

Comment:
I think it would be better just to increase the size of the buffer so that each set of results is included one after the other in the same buffer, rather than creating three separate buffers for each query.